bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/style/LabelAccumulator.java

```diff
@@ -72,13 +74,16 @@ public void accumulateConfigLabels(LabelStack configLabels, int columnPosition,
     }
 
     int columnIndex = bodyDataLayer.getColumnIndexByPosition(columnPosition);
-    ColumnType columnType = bodyDataProvider.getColumns().get(columnIndex).getColumnType();
-    if (columnType == ColumnType.SPAN_ANNOTATION) {
-      // Assign the span annotation style label to cells in columns with the respective flag.
-      configLabels.addLabel(StyleConfiguration.SPAN_ANNOTATION_CELL_STYLE);
-    } else if (columnType == ColumnType.TOKEN_TEXT) {
-      // Assign the token text style label to cells in columns with the respective flag.
-      configLabels.addLabel(StyleConfiguration.TOKEN_TEXT_CELL_STYLE);
+    List<Column> columns = bodyDataProvider.getColumns();
+    if (columns.size() != 0) {
```

Comment:
Size can be expensive to  calculate, e.g. for linked lists. If you just want to know if something is not empty, use the `isEmpty()` method.
```suggestion
    if (!columns.isEmpty()) {
```

Reply:
Ah, done already :)